playbooks/roles/ocp_client/tasks/oc_install.yml

```diff
@@ -0,0 +1,85 @@
+- name: Make a request to trigger tools extraction
+  ansible.builtin.uri:
+    url: "{{ ocp_client_url.split('/')[:-1] | join('/') }}"
+    return_content: true
+
+- name: Remove OC client archive and checksum file if exist
+  ansible.builtin.file:
+    state: absent
+    path: "{{ item }}"
+  loop:
+    - "{{ ocp_client_archive_dest_dir }}/{{ ocp_client_archive_name }}"
+    - "{{ ocp_client_archive_dest_dir }}/{{ checksum_file_name }}"
+
+- name: Download checksum file
+  register: download_result
+  until: download_result is not failed
+  retries: 60
+  delay: 10
+  notify: "Delete checksum file"
+  ansible.builtin.get_url:
+    url: "{{ ocp_client_url.split('/')[:-1] | join('/') }}/{{ checksum_file_name }}"
+    dest: "{{ ocp_client_archive_dest_dir }}/{{ checksum_file_name }}"
+    mode: '0640'
+
+- name: Read checksum file and it's store content in data variable
+  ansible.builtin.command: "cat {{ ocp_client_archive_dest_dir }}/{{ checksum_file_name }}"
+  changed_when: false
+  register: data
+
+- name: Print client's archive sha256
+  vars:
+    checksum: "{{ data.stdout_lines | select('match', '.*' ~ ocp_client_url.split('/')[-1] ~ '$') }}"
+  ansible.builtin.debug:
+    msg: "sha256:{{ checksum[0].split(' ')[0] }}"
+
+- name: Download openshift-client-linux.tar.gz
+  register: download_result
+  until: download_result is not failed
+  retries: 60
+  delay: 10
+  notify: "Delete openshift-client-linux.tar.gz"
+  vars:
+    checksum: "{{ data.stdout_lines | select('match', '.*' ~ ocp_client_url.split('/')[-1] ~ '$') }}"
+  ansible.builtin.get_url:
+    url: "{{ ocp_client_url }}"
+    dest: "{{ ocp_client_archive_dest_dir }}/{{ ocp_client_archive_name }}"
+    mode: '0640'
+    checksum: "sha256:{{ checksum[0].split(' ')[0] }}"
+
+- name: Extract archive
+  ansible.builtin.unarchive:
+    src: "{{ ocp_client_archive_dest_dir }}/{{ ocp_client_archive_name }}"
+    dest: "{{ ocp_client_archive_dest_dir }}"
+    remote_src: true
+
+- name: Ensure user's .local/bin directory exists
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: directory
+    mode: "0755"
+  loop:
+    - "{{ ansible_env.HOME }}/.local"
+    - "{{ ansible_env.HOME }}/.local/bin"
+
+- name: Move oc and kubectl binaries to user's .local/bin
+  notify: "Delete oc and kubectl binaries from temp directory"
+  ansible.builtin.copy:
+    src: "{{ ocp_client_archive_dest_dir }}/{{ item }}"
+    dest: "{{ ansible_env.HOME }}/.local/bin/{{ item }}"
+    mode: '0774'
+    remote_src: true
+  loop:
+    - "oc"
+    - "kubectl"
+
+- name: Verify 'oc' binary by running 'oc version'
```

Comment:
Should we assert that .local/bin is in PATH?

Reply:
No it's OK. In previous step I removed OC from all possible locations.